### PR TITLE
162 align product card size in desktop view

### LINKED ISSUE #162: align product card size in desktop view
Stretch the product card size to the size of Grid component
image

### PR DESCRIPTION
**Pull request by ikuno815**
# Changes Made:
This PR improves the styling for screens more than medium size(675pt).
In the case that users see product cards in 2 columns, product__content, product__title, and product-avail-container should be in the same size regardless of text length.
before:
before

after: 
after

## Issue No(s).:

Resolves #162 

# Before Submitting:
- [x] I have included a clear description of the contents of my Pull Request.
- [x] I have linked the associated Issue(s) above.
  - "Resolves #[Issue No.]"
  - Include a new line for each Issue if more than one is covered.
  - If there is no associated Issue, this PR will not be approved.
- [x] I have included the appropriate labels.
  - They should match the labels for the associated Issue(s).

# After Submitting:
- One of the Oriori team will review your code.
- Changes may be requested before approval, so please be sure to keep an eye on the status of your PR.

### CODE REVIEW

File: client/src/styles/ProductCard.css [hunk 1]
```diff
@@ -128,4 +129,54 @@
...
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		position: absolute;
+	}
+
+	.product__avail-text {
+		white-space: pre-line;
+		font-size: medium;
+	}
```
Comment (kappanjoe) [inline client/src/styles/ProductCard.css:173]: Please remove (redundant)

File: client/src/styles/ProductCard.css [hunk 2]
```diff
@@ -128,4 +129,54 @@
...
+
+	.product__content {
+		display: flex;
+		flex-direction: row;
```
Comment (kappanjoe) [inline client/src/styles/ProductCard.css:177]: Please remove (redundant)

File: client/src/styles/ProductCard.css [hunk 3]
```diff
@@ -128,4 +129,54 @@
...
+	.product__card {
+		max-width: 360pt;
+		min-width: 300pt;
+		display: flex;
+		flex-direction: column;
+		justify-content: space-around;
+		align-items: center;
+		align-content: center;
+		margin: 4pt auto;
+	}
```
Comment (kappanjoe) [inline client/src/styles/ProductCard.css:144]: Please remove (redundant)

File: client/src/styles/ProductCard.css [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 	max-width: 360pt;
-	min-width: 300px;
+	min-width: 225pt;
 	display: flex;
 	flex-direction: column;
 	justify-content: space-around;
 	align-items: center;
 	align-content: center;
-  margin: 4pt auto;
+    margin: 4pt auto;
```
Comment (kappanjoe) [inline client/src/styles/ProductCard.css:9]: Thank you for fixing this spacing!

File: client/src/styles/ProductCard.css [hunk 5]
```diff
@@ -128,4 +129,54 @@
...
+		height: 50pt;
+		display: flex;
+		align-items: center;
+		justify-content: center;
+		position: relative;
+	}
+
+	.product__avail-container {
+		background-color: #E5E5E5;
+		border-radius: 6pt;
```
Comment (kappanjoe) [inline client/src/styles/ProductCard.css:158]: Please remove (redundant)

**Review by kappanjoe — CHANGES_REQUESTED** [review]
This looks a lot better, thank you!! There are a few lines of code that are redundant and can be removed, so I pointed them out in the review comments. Also, there are two small areas that could possibly be improved, if you wouldn't mind working a little bit more on them.

The first is that the availability text in the two-column view is not vertically centered similar to the one-column view.
Two-column view:
two-column view
One-column view: 
one-column view
It's a small difference but hopefully you can see how the text in the one-column view is more centered between the top and bottom of the availability banner.

The second is the gap on the left side of taller images in the two-column view:
image gap
I wonder if we can eliminate these gaps? I'm not sure if it's better to have the gap wider on the right side or if it's okay to crop the images a little with the `img-container` element so they fill up the entire width... Maybe @nikobatzi1990 has some thoughts about this?

File: client/src/styles/ProductCard.css (hunk 1 above)
Comment (ikuno815) [reply client/src/styles/ProductCard.css:173]: Thank you for pointing it out! I removed those.

**Comment by nikobatzi1990** [comment]
Hmmm, perhaps we could try changing the CSS so that the image fills the whole space that it's contained in? I think we did something similar on for the Sandwich Matching Mockup with some of the images being pretty long. If that doesn't work, we can always manually crop the images before uploading 🤔

**Comment by kappanjoe** [comment]
> Hmmm, perhaps we could try changing the CSS so that the image fills the whole space that it's contained in?

I think that would look the best so the images are consistent! If we do that, however, there are some images that have text that will get cropped. Is that something we're okay with? Or should we consider an alternative?

**Comment by nikobatzi1990** [comment]
I think it should be ok since the users can click on them and will see the whole image on the press release (or whatever page the image links to).

**Comment by ikuno815** [comment]
> Hmmm, perhaps we could try changing the CSS so that the image fills the whole space that it's contained in?
Thank you for your advice! "object-fit: cover" is used in the Matching Mockup.
image
This image is after adding a white background color to product__img-container and centering the photo. What do you think?
I think we could choose photos that only have the product itself in the case of such as Genshin Yaki.

**Comment by kappanjoe** [comment]
> image This image is after adding a white background color to product__img-container and centering the photo. What do you think? I think we could choose photos that only have the product itself in the case of such as Genshin Yaki.

Agreed! I think this looks great - I'll go ahead and approve.

**Review by kappanjoe — APPROVED** [review]

**Pull request merged**